### PR TITLE
Adds authorization:RoleAssignment replacements

### PR DESCRIPTION
**Pull request by jeffmorrell**
Adds authorization:RoleAssignment overrides for forcing replace
Fix https://github.com/pulumi/pulumi-azure-native/issues/771

### CODE REVIEW

File: provider/pkg/gen/replacement.go [hunk 1]
```diff
@@ -7,6 +7,9 @@ import "github.com/pulumi/pulumi/pkg/v3/codegen"
 // forceNewMap is a map of Azure Resource Provider -> Resource Name -> Properties that cause replacements.
 // API Versions are currently ignored.
 var forceNewMap = map[string]map[string]codegen.StringSet{
+	"Authorization": {
+		"RoleAssignment": codegen.NewStringSet("applicationId", "principalId", "scope"),
```
Comment (mikhailshilkov) [inline provider/pkg/gen/replacement.go:11]: According to [docs](https://www.pulumi.com/docs/reference/pkg/azure-native/authorization/roleassignment/) this resource has no property called `applicationId`

**Review by mikhailshilkov — COMMENTED** [review]
Thank you for the PR! Could you please add a short description to `CHANGELOG.md`?

File: provider/pkg/gen/replacement.go (hunk 1 above)
Comment (morrell) [reply provider/pkg/gen/replacement.go:11]: Interesting, was going off the error message I receive. I'll remove the applicationId as the other two are the only valid ones reported in the error message.

**Comment by mikhailshilkov** [comment]
Alright, the change LGTM but we need to run `make build` to propagate the changes to the metadata file. Would you be able to do so or should I?

**Comment by morrell** [comment]
> Alright, the change LGTM but we need to run `make build` to propagate the changes to the metadata file. Would you be able to do so or should I?

@mikhailshilkov if you wouldn't mind doing it I would very much appreciate it.

**Comment by mikhailshilkov** [comment]
I'm not allowed to push to your fork so I had to open https://github.com/pulumi/pulumi-azure-native/pull/1010

**Pull request closed without merging**